### PR TITLE
Set.at_exn

### PR DESCRIPTION
**Pull request by UnixJunkie**
If it is accepted, I can propose the version not throwing exceptions on top of it soon.

### CODE REVIEW

File: src/batSet.ml [hunk 1]
```diff
@@ -793,6 +818,18 @@ let remove x s = Concrete.remove Pervasives.compare x s
 
 let iter f s = Concrete.iter f s
 
+let at_exn i s = Concrete.at_exn i s
+
+(*$T
+  at_exn 0 (of_list [1;2]) == 1
+  at_exn 1 (of_list [1;2]) == 2
```
Comment (gasche) [inline src/batSet.ml]: I think you could have a test with a bit more elements in the set -- for example with 5 elements, picking an element that is neither the first nor the last to avoid edge cases.

File: src/batSet.mli [hunk 2]
```diff
@@ -121,6 +121,12 @@ sig
       The elements of [s] are presented to [f] in increasing order
       with respect to the ordering over the type of the elements. *)
 
+  val at_exn: int -> t -> elt
+  (** [at_exn i s] returns element at rank [i] in [s].
```
Comment (gasche) [inline src/batSet.mli]: You might want to be a bit more explicit about what you mean by `at rank [i]` -- that it is about the ordering.

**Comment by UnixJunkie** [comment]
All done.

**Comment by gasche** [comment]
I think we want a similar function in `Map` for consistency, and this suggests that the name may be problematic: if we use `Map.at`, I think users would easily confuse it with `find`: you would expect to pass a key rather than an integer.

I think that `at_rank` may thus be a better name for this feature. More precisely, this version would be `at_rank_exn`. What do you think?

**Comment by UnixJunkie** [comment]
I'm fine with renaming: you know the code base more than me, if it's a good name for you it will be fine with me.

**Comment by UnixJunkie** [comment]
The feature was accepted from another PR.
